CMakeLists.txt

```diff
@@ -182,7 +195,7 @@ if(CMAKE_COMPILER_IS_CLANG)
     SET(CMAKE_C_FLAGS "-Qunused-arguments")
   endif()
   if("${CMAKE_CXX_FLAGS}" STREQUAL "")
-    SET(CMAKE_CXX_FLAGS "-ftemplate-depth=1024 -Qunused-arguments -Wno-invalid-offsetof ${SSE_FLAGS}") # Unfortunately older Clang versions do not have this: -Wno-unnamed-type-template-args
+    SET(CMAKE_CXX_FLAGS "-Wall -Wextra -ftemplate-depth=1024 -Qunused-arguments -Wno-invalid-offsetof ${SSE_FLAGS}") # Unfortunately older Clang versions do not have this: -Wno-unnamed-type-template-args
```

Comment:
So on my local machine, with only the SSE flags active, the compiler output seems to be pretty clean so far.

The -Qunused-argument is necessary only because of a bug in ccache's interaction with clang (which eventually was fixed, but not on the version included in precise). So we can set it conditionally on the value of the TRAVIS env (which is set to true on travis builds).